source/features/inline-instagram-photos.js

```diff
@@ -0,0 +1,46 @@
+import {h} from 'dom-chef';
+
+const instagramUrls = new Map();
+
+function createPhotoElement(imageUrl, postUrl) {
+	return <div class="AdaptiveMediaOuterContainer">
+			<div class="AdaptiveMedia">
+				<a href={`${postUrl}`} target="_blank" rel="noopener">
+					<img class="refined-twitter_instagram-inline" src={`${imageUrl}`} />
+				</a>
+			</div>
+		</div>;
+}
+
+async function getInstagramPhotoUrl(postUrl) {
+	const imageRegex = /"display_url": ?"([^"]+)"/;
+	const response = await fetch(postUrl);
+	const html = await response.text();
+	const [, instagramImageUrl] = imageRegex.exec(html) || [];
+	return instagramImageUrl;
+}
+
+export default function () {
```

Comment:
To remove the duplication, you can just check whether it contains the class `refined-twitter_instagram-inline`. If it does, it already inlined the image.

Reply:
Currently the class is added only to the element that contains the image and doesn’t exists before the XHR complete, but you gave me an idea: how about I add the element as soon as I find the twitter link and just insert the image when the request completes? That way, the old sibling check would work as expected and prevent the race condition.

Reply:
I meant “instagram link” :P

Reply:
Sounds good :)